lib/src/queue_list.dart

```diff
@@ -84,10 +102,10 @@ class QueueList<E> extends Object with ListMixin<E> implements Queue<E> {
     if (this is QueueList<T>) {
```

Comment:
While you are here, please rewrite this as:
```
QueueList<Object> self = this;
return self is QueueList<T> ? self : this.retype<T>();
```
It generates better code for dart2js to not have an explicit cast. By up-casting first. we make sure that a successful test promotes the variable.

Ditto for the `cast` methods below.

Reply:
Done.